consensus/src/liveness/leader_reputation.rs

```diff
@@ -612,26 +622,49 @@ impl LeaderReputation {
                 )
                 .collect();
 
-                CHAIN_HEALTH_PARTICIPATING_VOTING_POWER[counter_index].set(
-                    candidates
-                        .iter()
-                        .zip(self.voting_powers.iter())
-                        .filter(|(c, _vp)| participants.contains(c))
-                        .map(|(_c, vp)| *vp as f64)
-                        .sum(),
-                );
+                let participating_voting_power = candidates
+                    .iter()
+                    .zip(self.voting_powers.iter())
+                    .filter(|(c, _vp)| participants.contains(c))
+                    .map(|(_c, vp)| *vp as f64)
+                    .sum();
+
+                CHAIN_HEALTH_PARTICIPATING_VOTING_POWER[counter_index]
+                    .set(participating_voting_power);
                 CHAIN_HEALTH_PARTICIPATING_NUM_VALIDATORS[counter_index]
                     .set(participants.len() as i64);
+
+                if chosen {
+                    // do not treat chain as unhealthy, if chain just started, and we don't have enough history to decide.
+                    result = Some(
+                        if history.len() < *participants_window_size && self.epoch <= 2 {
+                            1.0
+                        } else {
+                            participating_voting_power / total_voting_power
```

Comment:
Is there any case where `total_voting_power == 0` could cause division by zero? (e.g. If there is no quorum?) I can see there has to be at least one entry in voting_powers but can't tell if that is guaranteed to be non-zero.